src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -503,7 +504,11 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
 
             this.RaiseCompilationTaskStart("OverallCompilation", "Build");
             this.compilationStatus.Validation = Status.Succeeded;
-            var files = CompilationUnitManager.InitializeFileManagers(sourceFiles, null, this.OnCompilerException); // do *not* live track (i.e. use publishing) here!
+            var files = CompilationUnitManager.InitializeFileManagers(
+                sourceFiles,
+                null,
+                this.OnCompilerException,  // do *not* live track (i.e. use publishing) here!
+                degreeOfParallelism: CompilationBuilder.Utils.IsWebAssembly == true ? 1 : (int?)null);

```

Comment:
perhaps it would make more sense to add disabling of parallelism as option to `Configuration` instead and then let the caller decide (by sniffing out the runtime - same as here, or any other mechanism) whether it's needed or not. This would allow to run test against the disable parallelization easily, also from outside of WASM, which I think would be beneficial as it would increase the confidence in the compiler working correctly in such mode too